fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryFirstOrDefaultPlan.java

```diff
@@ -96,10 +96,19 @@ public <M extends Message> RecordCursor<QueryResult> executePlan(@Nonnull final
                                                                      @Nonnull final EvaluationContext context,
                                                                      @Nullable final byte[] continuation,
                                                                      @Nonnull final ExecuteProperties executeProperties) {
-        return new FutureCursor<>(store.getExecutor(),
-                getChild().executePlan(store, context, continuation, executeProperties).first()
-                        .thenApply(resultOptional ->
-                                resultOptional.orElseGet(() -> QueryResult.ofComputed(onEmptyResultValue.eval(store, context)))));
+        if (continuation == null) {
+            return new FutureCursor<>(store.getExecutor(),
```

Comment:
Is now the right time to add some tests of `FutureCursor` directly? It doesn't look like it has any unit tests.

Reply:
Maybe... I'm not convinced that there's much we're going to get out of just those tests, given that the tricky part is more about how it's used. I was thinking if there should be a method like:

```java
static RecordCursor<T> futureCursor(Supplier<CompletableFuture<T>> supplier, @Nullable byte[] continuation, Executor executor) {
   return continuation == null ? new FutureCursor(executor, supplier.get()) : RecordCursor.empty(executor);
}
```

And then we could assert that _that_ did the right thing...ish.

Reply:
I added some basic unit tests to `RecordCursorTest`. We also already had a `fromFuture` method that seems like it was trying to do much of the same thing as this logic here, so I switched over to it. I added some verbiage in the javadoc to reflect things we discussed in this PR review